Add toString() method

igreenfield opened the pull request.

Commit:
Add toString() method

Commit:
Add toString() method

allanbank commented:
Would you mind changing this to match the format of the upcoming '[find](https://github.com/mongodb/specifications/blob/master/source/find_getmore_killcursors_commands.rst#find)' command? 

It is also nice to have the same toString() on the Builder class.

Commit:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/com/allanbank/mongodb/builder/Find.java

igreenfield commented:
@allanbank Are you intend to merge this?

Commit:
Fix the built-in "majority" mode

allanbank commented:
Short answer is yes.

We have a rather large change to the driver that we have been using internally that we have not been able to find the time to cleanup, put through the test gauntlet, and release. I am hopeful an intern will be able to get that done early summer.

Commit:
do not set 'myNumberToReturn' to negative number

There have been a number of significant changes in the product that were introduced in versions 3.0 and 3.2 that were released in 2015.  One such change is enhancements to our wire protocol.  Previously the legacy `OP_QUERY` spec had a single parameter, `ntoreturn`, which was overloaded to be used with both `limit` and `batchSize` options.  This behavior is described in our [legacy CRUD API specification](https://github.com/mongodb/specifications/blob/master/source/crud/crud.rst#combining-limit-and-batch-size-for-the-wire-protocol).  The ambiguity in terms of what the client was expecting when `ntoreturn` was provided resulted in suboptimal handling of the request in some situations.

Commit:
revert until I fix all the tests

Commit:
Fix tests for mongo3.2

Commit:
Ignore failed tests

Commit:
Fix numberToReturn, Fix tests.

Commit:
Delete mongodb-async-driver.iml

Commit:
Ignore failed tests

Commit:
Add after class log

Commit:
Add logging to start and stop tests

Commit:
improve test logging.

Commit:
Fix test
Change pom version

Commit:
Add logging to test

Commit:
set reuseForks to false

Commit:
set <forkMode>always</forkMode>

Commit:
Add message to assert

Commit:
Fix URL

Commit:
Fix explain test

Commit:
Add log to assert

Commit:
Add if on test env

Commit:
use forkCount and reuseForks in maven-surefire-plugin

Commit:
remove local variable.

Commit:
Add condition to if in test.

Commit:
Add logs

Commit:
Change log level to INFO

Commit:
Add logging.

Commit:
Ignore some tests

Commit:
add if NPE

Commit:
Fix auth test

Commit:
add if NPE

Commit:
Add delete of mongo lock file

Commit:
remove import

Commit:
increase time in test

Commit:
remove set parameter

Commit:
Add logging

Commit:
Change profiling level

Commit:
Fix delete file path

Commit:
add wait aftert stop.

Commit:
Move to JDK 1.8

Commit:
Change log level of tetsts

Commit:
Change version

Commit:
JUL level to ALL

Commit:
change log for surefire

Commit:
Add stop process method

Commit:
sleep 15 seconds

Commit:
Run only one Test

Commit:
sleep more time

Commit:
Fix test

Commit:
Check for ENV also

Commit:
Change repair logic

Commit:
Add logging...

Commit:
add --nojournal

Commit:
enable mongodb logs

Commit:
pkill -9

Commit:
new plugin versions

Commit:
Add prefix to mongo output

Commit:
Add logging

Commit:
Add logging

Commit:
remove unneeded line

Commit:
Change rs configuration

Commit:
Change the mongo connection

Commit:
Some fixes

Commit:
Some fixes

Commit:
add --bind_ip 127.0.0.1

Commit:
change the --bind_ip tp 0.0.0.0

Commit:
Add regex
